components/proxy/src/main/java/com/hotels/styx/StyxConfig.java

```diff
@@ -19,62 +19,43 @@
 import com.hotels.styx.admin.AdminServerConfig;
 import com.hotels.styx.api.Resource;
 import com.hotels.styx.api.configuration.Configuration;
-import com.hotels.styx.common.io.ResourceFactory;
 import com.hotels.styx.client.StyxHeaderConfig;
+import com.hotels.styx.common.io.ResourceFactory;
+import com.hotels.styx.config.schema.SchemaValidationException;
 import com.hotels.styx.infrastructure.configuration.ConfigurationParser;
 import com.hotels.styx.infrastructure.configuration.yaml.YamlConfiguration;
 import com.hotels.styx.proxy.ProxyServerConfig;
+import org.slf4j.Logger;
 
 import java.nio.file.Path;
-import java.nio.file.Paths;
 import java.util.Optional;
 
-import static com.hotels.styx.StartupConfig.defaultStartupConfig;
+import static com.hotels.styx.ServerConfigSchema.validateServerConfiguration;
 import static com.hotels.styx.infrastructure.configuration.ConfigurationSource.configSource;
 import static com.hotels.styx.infrastructure.configuration.yaml.YamlConfigurationFormat.YAML;
+import static java.lang.System.getProperty;
 import static java.util.Objects.requireNonNull;
 import static java.util.stream.Collectors.toList;
 import static java.util.stream.StreamSupport.stream;
+import static org.slf4j.LoggerFactory.getLogger;
 
 /**
  * Styx configuration.
  */
 public final class StyxConfig implements Configuration {
+    private static final Logger LOG = getLogger(StyxConfig.class);
+    private static final String VALIDATE_SERVER_CONFIG_PROPERTY = "validateServerConfig";
+
     public static final String NO_JVM_ROUTE_SET = "noJvmRouteSet";
 
-    private final StartupConfig startupConfig;
     private final Configuration configuration;
 
-    private final ProxyServerConfig proxyServerConfig;
-    private final AdminServerConfig adminServerConfig;
-    private final StyxHeaderConfig styxHeaderConfig;
-
     public StyxConfig() {
-        this(defaultStartupConfig(), EMPTY_CONFIGURATION);
+        this(EMPTY_CONFIGURATION);
     }
 
     public StyxConfig(Configuration configuration) {
```

Comment:
I understood we were going to hide these constructors so we can make sure that configuration is validated if necessary.

Reply:
No we are just deleting some constructors that are no longer needed since we moved YAML parsing to a static method, and extracted `StartupConfig` to be handled separately.

Reply:
Ok... I would still prefer a model in which we can validate any "Configuration" (if the validation is implemented!) but since we don't have any such validation it might be a bit redundant. Setting this to resolved.